airbyte-ci/connectors/pipelines/pipelines/models/reports.py

```diff
@@ -70,7 +74,7 @@ def skipped_steps(self) -> List[StepResult]:
 
     @property
     def success(self) -> bool:
-        return len(self.failed_steps) == 0 and (len(self.skipped_steps) > 0 or len(self.successful_steps) > 0)
+        return len(self.considered_failed_steps) == 0 and (len(self.skipped_steps) > 0 or len(self.successful_steps) > 0)
```

Comment:
Edge case: if we only return failed and ignored steps, this doesn't return true (because no steps were successfull/skipped). don't think we need to worry about this too much, but we're definitely getting out of our enum groups and into more complicated stuff

Reply:
> My ask is about variable naming clarity. It feels to me that the intent behind consider_result_in_overall_status is basically is_hard_failure.

> I.e. if it's a hard fail (consider in overall result == true), fail the pipeline if this is failed.
If it's not a hard fail, than only fail the pipeline if the corresponding check on master does not fail.


This is interesting - we could even pull this back into our enums, instead of creating a second option, by creating 
`StepStatus.SOFT_FAILURE` and StepStatus.HARD_FAILURE` for example

Reply:
I like this new enum strategy but it'd would require a significant refacto to change the failure detection everywhere it's used. I suggest doing it if we have multiple "soft failure" implementation coming up.

Reply:
> Edge case: if we only return failed and ignored steps, this doesn't return true (because no steps were successfull/skipped). don't think we need to worry about this too much, but we're definitely getting out of our enum groups and into more complicated stuff

I remember that I originally set this to `return not self.failed_steps` (which we could transform to `return not self.considered_failed_steps`) . I remember  I added the `and`  to fix another edge case - I should have added a comment to remember which one!